Fixes for list / create deployment in self hosted mode

argha-dot opened the pull request:
- moved workspace sidebar card to it's own component
 - setting dummy 'last_used_workspace_id' to `Uuid::nil()` when the app is self-hosted
 - added `AppType` struct to determine if the app is self-hosted or managed
 - conditionally rendering parts of login, sign up, create deployment, and deployment list pages depending on the `AppType`

argha-dot committed:
- moved workspace sidebar card to it's own component
 - setting dummy 'last_used_workspace_id' to `Uuid::nil()` when the app is self-hosted
 - added `AppType` struct to determine if the app is self-hosted or managed
 - conditionally rendering parts of login, sign up, create deployment, and deployment list pages depending on the `AppType`

rakshith-ravi commented on hosted-frontend/src/pages/auth/login.rs:
```diff
@@ -63,7 +64,7 @@ pub fn LoginForm() -> impl IntoView {
 						access_token,
 						refresh_token,
-						last_used_workspace_id: None,
+						last_used_workspace_id: Some(Uuid::nil()),
```
Shouldn't this be decided based on the app_type?

argha-dot replied:
Missed this, I'll put a fix for this along with the next review in the next commit?,

rakshith-ravi commented on hosted-frontend/src/pages/auth/sign_up.rs:
```diff
@@ -85,7 +86,12 @@ pub fn SignUpForm() -> impl IntoView {
 
 			match response {
-				Ok(CreateAccountResponse {}) => {}
+				Ok(CreateAccountResponse {}) => {
```
Same - should be based on app_type

rakshith-ravi commented on runners/common/src/routes/workspace/deployment/list_deployment.rs:
```diff
@@ -54,10 +54,9 @@ pub async fn list_deployment(
-			let image_tag = row.try_get::<String, _>("image_tag")?;
 			let status = row.try_get::<DeploymentStatus, _>("status")?;
 			let registry = row.try_get::<String, _>("registry")?;
-			let repository_id = row.try_get::<Uuid, _>("repository_id")?;
```
Why remove this?

argha-dot replied:
Right now, there's no repository_id row in the deployments table. I thought we would only support external registries.
I can add the repository_id to everything if that's not the case

rakshith-ravi replied:
Hmm, repository ID usually references Patr's repository table. I suppose since we don't have that table, we can get away with just image name. Should be fine

rakshith-ravi committed:
Refactor imports in db module and remove unused code

argha-dot committed:
- WIP update deployment details
 - fixes issue with conditional setting of worksapce_id and
   navigating on sign-up
 - fixed sql query
 - fixed minor css issues

rakshith-ravi commented on hosted-frontend/src/queries/infrastructure/deployment/mod.rs:
```diff
@@ -216,3 +215,44 @@ pub fn get_deployment_logs_query(
 }
+
+/// Query to update a deployment, Returns an action to be dispatched on submit.
+pub fn udpdate_deployment_query() -> Action<
```
Small typo

argha-dot replied:
Fixed the typo, in a commit which has a typo

rakshith-ravi replied:
GG

rakshith-ravi reviewed (dismissed):
One small typo. The rest of it looks good to me

rakshith-ravi commented:
Not sure why this is marked as draft. Lmk when this is good to go

argha-dot committed:
fix type
